File: solutions/java/resistor-color/1/src/main/java/ResistorColor.java
import java.util.stream.Stream;

class ResistorColor {

    static final String[] colors = { "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white" };

    int colorCode(String color) {
        for (int value = 0; value < 10; value++)
            if (colors()[value].equals(color))
                return value;
        
        return -1;
    }

    String[] colors() {
        return colors;
    }
}

File: solutions/typescript/resistor-color/1/resistor-color.ts
export const colorCode = (color:string) => {

  for (let index = 0; index < COLORS.length; index++)
    if (COLORS[index] == color)
      return index;

}

export const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
]
